src/lib/viewers/box3d/image360/Image360Viewer.js

```diff
@@ -30,9 +33,46 @@ class Image360Viewer extends Box3DViewer {
      * @inheritdoc
      */
     createSubModules() {
-        this.controls = new Box3DControls(this.wrapperEl);
+        this.controls = this.getViewerOption('useReactControls')
+            ? new ControlsRoot({ containerEl: this.wrapperEl, fileId: this.options.file.id })
+            : new Box3DControls(this.wrapperEl);
         this.renderer = new Image360Renderer(this.wrapperEl, this.boxSdk, { api: this.api });
     }
+
+    /**
+     * @inheritdoc
+     */
+    handleSceneLoaded() {
+        super.handleSceneLoaded();
+
+        this.renderUI();
+    }
+
+    /**
+     * @inheritdoc
+     */
+    handleShowVrButton() {
+        if (this.controls && this.getViewerOption('useReactControls')) {
```

Comment:
Should the `this.controls` identity check be a guard clause? Otherwise the `else` clause below could throw.